nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -0,0 +1,310 @@
+    """
+    Returns `ranking_subset_size` indices of elements in the `errors` list
+    that have the biggest error value. Returned indices are sorted in
+    ascending order.
+
+    :param errors: A list of errors.
+    :param ranking_subset_size: A number of returned indices.
+    :return: Indices of elements in the `errors` list which have the biggest error value.
+    """
+    ordered_indices = [
+        idx for idx, _ in sorted(enumerate(errors), key=operator.itemgetter(1), reverse=True)
+    ]
+    end_index = min(ranking_subset_size, len(ordered_indices))
+    return sorted(ordered_indices[:end_index])
+
+
+@dataclass
+class GroupToRank:
+    """
+    Describes a group of quantizers to rank.
+
+    :param quantizers: List of quantizer nodes.
+    :param operations: List of nodes that will be reverted to original precision
+        if `GroupToRank.quantizers` are removed.
+    """
+
+    quantizers: List[NNCFNode]
+    operations: List[NNCFNode]
+
+    # TODO(andrey-churkin): Add hash
+
+
+class Ranker(ABC):
+    """
+    Encapsulates logic to rank groups of quantizers.
+    """
+
+    def __init__(self,
+                 ranking_subset_size: int,
+                 ranking_fn: Callable[[Any, Any], float],
+                 dataset: Dataset,
+                 algo_backend: AccuracyControlAlgoBackend):
+        """
+        :param ranking_subset_size: The number of data items that will be selected from
+            the dataset to rank groups of quantizers. The `len(dataset)` data items will
+            be selected if `ranking_subset_size` parameter is greater than the number of
+            elements in the dataset.
+        :param ranking_fn: A function that compares values returned by
+            `_collect_values_for_each_item()` for initial and quantized models.
+        :param dataset: Dataset for the ranking process.
+        :param algo_backend: The `AccuracyControlAlgoBackend` algo backend.
+        """
+        self._ranking_subset_size = ranking_subset_size
+        self._ranking_fn = ranking_fn
+        self._dataset = dataset
+        self._algo_backend = algo_backend
+        # We don't need to re-calculate values for the initial model
+        # because they don't change. So use this attribute to store
+        # them to improve execution time.
+        self._ref_values = None
+
+    def find_groups_of_quantizers_to_rank(self, quantized_model_graph: NNCFGraph) -> List[GroupToRank]:
+        """
+        Finds groups of quantizers to rank.
+
+        :param quantized_model_graph: Graph for quantized model.
+        :return: List of groups of quantizers to rank.
+        """
+        groups_to_rank = []
+        processed = {}
+        # TODO(andrey-churkin): Set order of quantizers here.
+        quantizers = quantized_model_graph.get_nodes_by_metatypes(self._algo_backend.get_quantizer_metatypes())
+        quantized_model_graph_without_shapeof = remove_shapeof_subgraphs(
```

Comment:
Why not use `transform_to_inference_graph`?

Reply:
Actually, I don't understand why the `transform_to_inference_graph()` method is needed. It just calls the `remove_shape_of_subgraph()`. So I expect that `transform_to_inference_graph()` will be extended in the future. Otherwise, it should be removed, and `remove_shape_of_subgraphs()` should be used.

I need to remove ShapeOf subgraphs and I call the method which does exactly what I need. Where is the problem?

Reply:
It may be extended in the future, but not necessarily.
If you need to get an `inference graph` without the `shape of subgraphs`, why not use a more common method for this?

Reply:
I don't need an inference graph. I need a graph without ShapeOf subgraphs.

Reply:
Ok, as you wish.
For now, there is no difference between an `inference` graph or a graph without a `shape of.